docs/design/ovs-pipeline.md

```diff
@@ -333,58 +382,195 @@ more information on connection tracking in OVS.
+`service.spec.sessionAffinityConfig.clientIP.timeoutSeconds` of the Service will be
+generated in this table. This will be explained in detail in chapter table [ServiceLBTable].
+
+### ServiceLBTable (41)
+
+This table is used to implement Service Endpoint selection. Note that, currently, only
+ClusterIP whose client is from the cluster is supported, which means that client should
+be Pods. NodePort, LoadBalancer and ClusterIP whose client is from K8s Node will be
+supported in the future.
+
+When a ClusterIP Service whose `service.spec.sessionAffinity` is None is created, if you
+dump the flows, you may see the following one:
 
 ```text
-1. table=40, priority=200,ip,nw_dst=10.96.0.0/12 actions=load:0x2->NXM_NX_REG1[],load:0x1->NXM_NX_REG0[16],goto_table:105
-2. table=40, priority=0 actions=goto_table:45
+1. table=41, priority=200,tcp,reg4=0x10000/0x70000,nw_dst=10.107.100.231,tp_dst=443 actions=load:0x2->NXM_NX_REG4[16..18],load:0x1->NXM_NX_REG0[19],group:5
 ```
 
-In the example above, 10.96.0.0/12 is the Service CIDR (this is the default
-value used by `kubeadm init`). This flow is not actually required for
-forwarding, but to bypass [EgressRuleTable] and [EgressDefaultTable] for Service
-traffic on its way to kube-proxy through the gateway. If we omitted this flow,
-such traffic would be unconditionally dropped if a Network Policy is applied on
-the originating Pod. For such traffic, we instead enforce Network Policy egress
-rules when packets come back through the gateway and the destination IP has been
-rewritten by kube-proxy (DNAT to a backend for the Service). We cannot output
-the Service traffic to the gateway port directly as we haven't committed the
-connection yet; instead we store the port in NXM_NX_REG1 - similarly to how we
-process non-Service traffic in [L2ForwardingCalcTable] - and forward it to
-[ConntrackCommitTable]. By committing the connection we ensure that reply
-traffic (traffic from the Service backend which has already gone through
-kube-proxy for source IP rewrite) will not be dropped because of Network
-Policies.
+The match conditions of above flow:
+
+* `reg4=0x10000/0x70000`, value of bits [16..18] in NXM_NX_REG4 is 0b001, which is used
+  to match the Service packet whose state is to do Endpoint selection. The bits [16..18]
+  in NXM_NX_REG4 is set in table [SessionAffinityTable] by flow `table=40, priority=0 actions=load:0x1->NXM_NX_REG4[16..18]`.
+* `nw_dst=10.107.100.231` is used to match the IP address of ClusterIP.
+* `tcp` is used to match TCP protocol.
+* `tp_dst=443` is used to match the port number of ClusterIP.
+
+The actions of above flow:
+
+* `load:0x2->NXM_NX_REG4[16..18]` is used to set the value of bits [16..18] in NXM_NX_REG4
+  to 0b002, which indicates that Endpoint selection has been done. Note that, the Endpoint
+  selection has not really been done yet. Endpoint selection will be done in the target
+  OVS group. More strictly speaking, above action should be done in the OVS group entry.
```

Comment:
What does this sentence mean? I feel you should change "More strictly speaking" to something else.

Reply:
Thanks, updated.